File: python/adhoc/app/models/Feedback.py
from orator import Model

class Feedback(Model):
    __table__ = 'feedback'
    __timestamps__ = False

    def validate(self):
        if "@" in self.email:
            return True
        return False

    def save(self, **kwargs):
        if self.validate():
            return super().save()
        return False

File: php/adhoc/app/models/Feedback.php
<?php
namespace app\models;
use Illuminate\Database\Eloquent\Model;
class Feedback extends Model
{
    protected $table = 'feedback';
    public $timestamps = false;
    protected $fillable = ['nome', 'email', 'feedback'];
    public ?string $error = null;
    public function validate(): bool
    {
        if (strpos($this->email, '@') !== false) {
            return true;
        }

        $this->error = 'Email deve conter @';
        return false;
    }

    public function save(array $options = []): bool
    {
        if ($this->validate()) {
            // Chama o método original save() da classe pai (Eloquent\Model).
            return parent::save($options);
        }

        return false;
    }

}